e2e/toast/toast.e2e-spec.ts

```diff
@@ -0,0 +1,65 @@
+import {ToastPO} from '@agnos-ui/page-objects';
+import {ToastDemoPO} from '../demo-po/toast.po';
+import {expect, test} from '../fixture';
+
+type PromiseValue<T> = T extends Promise<infer U> ? U : never;
+type State = PromiseValue<ReturnType<ToastPO['state']>>;
+
+test.describe(`Toast tests`, () => {
+	test(`Dynamic toast should apply proper position`, async ({page}) => {
+		const toastDemoPO = new ToastDemoPO(page);
+
+		await page.goto('#/toast/dynamic');
+		await toastDemoPO.locatorRoot.waitFor();
+
+		await toastDemoPO.locatorPositionSelect.selectOption('middleCenter');
+		await toastDemoPO.locatorAddToastButton.click();
+
+		const middleCenterContainer = toastDemoPO.locatorToastContainer(4);
+		await expect(middleCenterContainer).toHaveClass(/top-50/);
+		await expect(middleCenterContainer).toHaveClass(/start-50/);
+		await expect(middleCenterContainer).toHaveClass(/translate-middle/);
+
+		await toastDemoPO.locatorPositionSelect.selectOption('topLeft');
+		await toastDemoPO.locatorAddToastButton.click();
+
+		const topLeftContainer = toastDemoPO.locatorToastContainer(0);
+		await expect(topLeftContainer).toHaveClass(/top-0/);
+		await expect(topLeftContainer).toHaveClass(/start-0/);
+	});
+
+	test(`Toast without header`, async ({page}) => {
+		const toastDemoPO = new ToastDemoPO(page);
+		await page.goto('#/toast/playground#{"props":{"slotDefault":"This is a toast","autohide":false,"className":"text-bg-primary"}}');
+		await toastDemoPO.locatorRoot.waitFor();
+		const toastPO = new ToastPO(page);
+
+		const expectedState: State = {
+			rootClasses: ['au-toast', 'd-flex', 'show', 'text-bg-primary', 'toast', 'toast-dismissible'],
+			body: 'This is a toast',
+			header: undefined,
+		};
+
+		await expect.poll(() => toastPO.state()).toEqual(expectedState);
+		await toastPO.locatorCloseButton.click();
+		await toastPO.locatorRoot.waitFor({state: 'hidden'});
+	});
+
+	test(`Toast with header and without close button`, async ({page}) => {
+		const toastDemoPO = new ToastDemoPO(page);
+		await page.goto(
+			'#/toast/playground#{"props":{"slotDefault":"This is a toast","autohide":false,"className":"text-bg-primary", "slotHeader": "Header", "dismissible": false}}',
+		);
+		await toastDemoPO.locatorRoot.waitFor();
+		const toastPO = new ToastPO(page);
+
+		const expectedState: State = {
+			rootClasses: ['au-toast', 'show', 'text-bg-primary', 'toast'],
+			body: 'This is a toast',
+			header: 'Header',
+		};
+
+		await expect.poll(() => toastPO.state()).toEqual(expectedState);
+		await expect(toastPO.locatorCloseButton).toHaveCount(0);
```

Comment:
Maybe you can continue the test by changing the dismissible and check that teh close button is here

Reply:
not sure that the test should modify the dismissible on the fly as it is not an expected behavior of the component

Reply:
no, i mean that i think it is susfficient to test whether it is there or not in different tests as users are not supposed to remove/add button on the fly

Reply:
Yes, you're right ;)